parquet/src/main/java/org/apache/iceberg/parquet/ParquetDictionaryRowGroupFilter.java

```diff
@@ -377,6 +381,31 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       return ROWS_CANNOT_MATCH;
     }
 
+    @Override
+    public <T> Boolean notStartsWith(BoundReference<T> ref, Literal<T> lit) {
+      int id = ref.fieldId();
+
+      Boolean hasNonDictPage = isFallback.get(id);
+      if (hasNonDictPage == null || hasNonDictPage) {
+        return ROWS_MIGHT_MATCH;
+      }
+
+      Set<T> dictionary = dict(id, lit.comparator());
+
+      // Allow query engine to make its own decisions regarding SQL 3-valued boolean logic.
+      if (dictionary.contains(null)) {
+        return ROWS_MIGHT_MATCH;
+      }
```

Comment:
The dictionary will never contain null, so you can remove this.

Reply:
Removed it.